app/controllers/api/work_orders_controller.rb

```diff
@@ -6,6 +6,21 @@ def description
 
     def notes_and_appointments
       @work_order = Hackney::WorkOrder.find(reference)
+
+      @notes_and_appointments = @work_order.notes + (@work_order.appointments.nil? ? [] : @work_order.appointments)
+    end
+
+    def notes
+      @work_order = Hackney::WorkOrder.find(reference)
+
+      if !params[:note][:text].empty?
```

Comment:
if you're checking empty here you might as well check for 2000. use `present?` instead of `!x.empty?` as it removes the negation and handles nil and blank strings.

Reply:
if there is an error here we should render some of message to the user